stock_shift_residual_qty/models/stock_picking.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2020 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def button_validate(self):
+        res = super().button_validate()
+        wh_ids = [x.picking_type_id.warehouse_id.id for x in self]
+        internal_types = self.env["stock.picking.type"].search(
+            [("warehouse_id", "in", wh_ids), ("code", "=", "internal")]
+        )
+        pick_types = {x.warehouse_id: x.id for x in internal_types}
+        if not pick_types:
+            raise UserError("No internal picking type available for any warehouse.")
+        for rec in self:
+            if rec._satisfy_condition_for_residual_picking() and pick_types.get(
+                rec.picking_type_id.warehouse_id
+            ):
+                vals = {
+                    "picking_type_id": pick_types[rec.picking_type_id.warehouse_id],
+                    "origin": "Residual from %s" % rec.name,
+                    "location_id": rec.location_id.id,
+                    "location_dest_id": rec.picking_type_id.residual_location_id.id,
+                    "move_lines": [
+                        (
+                            0,
+                            0,
+                            {
+                                "product_id": x["prd"].id,
+                                "name": x["prd"].name,
+                                "product_uom": x["prd"].uom_id.id,
+                                "product_uom_qty": x["qty"],
+                            },
+                        )
+                        for x in self._get_residual_product_and_qty()
```

Comment:
If "self._get_residual_product_and_qty()" is equal to {} it's create a empty picking ?

Reply:
You're right